lib/datadog/core/telemetry/logging.rb

```diff
@@ -5,13 +5,48 @@
 module Datadog
   module Core
     module Telemetry
-      # Logging interface for sending telemetry logs.
+      # === INTRENAL USAGE ONLY ===
+      #
+      # Logging interface for sending telemetry logs... so we can fix them.
+      #
+      # For developer using this module:
+      # - MUST NOT provide any sensitive information (PII)
+      # - SHOULD reduce the data cardinality for batching/aggregation
+      #
+      # Before using it, ask yourself:
+      # - Do we need to know about this (ie. internal error or client error)?
+      # - How severe/critical is this error? (ie. error, warning, fatal)
+      # - What information needed to make it actionable?
       #
-      # Reporting internal error so that we can fix them.
-      # IMPORTANT: Make sure to not log any sensitive information.
       module Logging
         extend self
 
+        # Extract datadog stack trace from the exception
+        module DatadogStackTrace
```

Comment:
I possibly not for this PR, but it would be cool to perhaps leave stack entries from:

* The Ruby standard library -- paths starting with `RbConfig::CONFIG.fetch("rubylibdir")` (The profiler uses this to categorize Ruby code automatically as being from the standard library!)
* dogstatsd-ruby -- by perhaps asking rubygems where dogstatsd-ruby is, when available